cmd/portal/commands/receive.go

```diff
@@ -0,0 +1,227 @@
+			if err := validateAddress(relayAddr); err != nil {
+				return fmt.Errorf("%w: (%s) is not a valid relay address", err, relayAddr)
+			}
+
+			logFile, err := setupLoggingFromViper("receive")
+			if err != nil {
+				return err
+			}
+			defer logFile.Close()
+
+			pwd := args[0]
+			if !password.IsValid(pwd) {
+				return fmt.Errorf("invalid password format")
+			}
+			switch viper.GetString("tui_style") {
+			case config.StyleRich:
+				if err := handleReceiveCommand(version, pwd); err != nil {
+					return fmt.Errorf("running rich receive command: %w", err)
+				}
+				return nil
+			case config.StyleRaw:
+				if err := handleReceiveCommandRaw(version, pwd); err != nil {
+					return fmt.Errorf("running raw receive command: %w", err)
+				}
+				return nil
+			default:
+				return errors.New("invalid tui style provided")
+			}
+		},
+	}
+	receiveCmd.Flags().StringP("relay", "r", "", relayFlagDesc)
+	receiveCmd.Flags().BoolP("yes", "y", false, "Overwrite existing files without [Y/n] prompts")
+	receiveCmd.Flags().StringP("tui-style", "s", "", tuiStyleFlagDesc)
+	return receiveCmd
+}
+
+// ------------------------------------------------------ Handlers -----------------------------------------------------
+
+// handleReceiveCommand is the receive application.
+func handleReceiveCommand(version string, password string) error {
+	var opts []receiver_tui.Option
+	ver, err := semver.Parse(version)
+	if err == nil {
+		opts = append(opts, receiver_tui.WithVersion(ver))
+	}
+	receiver := receiver_tui.New(viper.GetString("relay"), password, opts...)
+
+	if _, err := receiver.Run(); err != nil {
+		return fmt.Errorf("running receiver tui: %w", err)
+	}
+	fmt.Println("")
+	return nil
+}
+
+func handleReceiveCommandRaw(version string, password string) error {
+	ctx := context.Background()
+	relayAddr := viper.GetString("relay")
+	ver, err := semver.Parse(version)
+	if err != nil {
+		return fmt.Errorf("parsing version: %w", err)
+	}
+	serverVer, err := semver.GetRendezvousVersion(ctx, relayAddr)
+	if err != nil {
+		return fmt.Errorf("fetching version from relay: %w", err)
+	}
+	if ver.Compare(serverVer) == semver.CompareOldMajor {
+		return fmt.Errorf("incompatible version %s -> %s", ver, serverVer)
+	}
+	cnf := portal.Config{
+		RendezvousAddr: relayAddr,
+	}
+	temp, err := os.CreateTemp(os.TempDir(), file.RECEIVE_TEMP_FILE_NAME_PREFIX)
+	if err != nil {
+		return fmt.Errorf("creating temp receiver file: %w", err)
+	}
+
+	if err := portal.Receive(ctx, temp, password, &cnf); err != nil {
+		return fmt.Errorf("receiving files: %w", err)
+	}
+
+	if _, err := temp.Seek(0, 0); err != nil {
+		return fmt.Errorf("seeking to start of temp file: %w", err)
+	}
+	unpacker := file.NewUnpacker(viper.GetBool("prompt_overwrite_files"))
+	defer unpacker.Close()
+	defer file.RemoveTemporaryFiles(file.RECEIVE_TEMP_FILE_NAME_PREFIX)
+
+	if err := unpacker.Init(temp); err != nil {
+		return fmt.Errorf("initialising unpacker: %w", err)
```

Comment:
initialising -> initializing (to keep consistent with American English spelling)